client/landing/stepper/declarative-flow/assembler-first-flow.ts

```diff
@@ -0,0 +1,264 @@
+import { Onboard } from '@automattic/data-stores';
+import { DEFAULT_ASSEMBLER_DESIGN, isAssemblerSupported } from '@automattic/design-picker';
+import { useLocale } from '@automattic/i18n-utils';
+import { ASSEMBLER_FIRST_FLOW } from '@automattic/onboarding';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { getLocaleFromQueryParam, getLocaleFromPathname } from 'calypso/boot/locale';
+import { useQueryTheme } from 'calypso/components/data/query-theme';
+import { skipLaunchpad } from 'calypso/landing/stepper/utils/skip-launchpad';
+import { getCurrentUserSiteCount, isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getTheme } from 'calypso/state/themes/selectors';
+import { useSiteData } from '../hooks/use-site-data';
+import { ONBOARD_STORE, SITE_STORE } from '../stores';
+import { useLoginUrl } from '../utils/path';
+import { recordSubmitStep } from './internals/analytics/record-submit-step';
+import { STEPS } from './internals/steps';
+import { ProcessingResult } from './internals/steps-repository/processing-step/constants';
+import {
+	AssertConditionResult,
+	AssertConditionState,
+	Flow,
+	ProvidedDependencies,
+} from './internals/types';
+import type { OnboardSelect } from '@automattic/data-stores';
+
+const SiteIntent = Onboard.SiteIntent;
+
+const assemblerFirstFlow: Flow = {
+	name: ASSEMBLER_FIRST_FLOW,
+	useSideEffect() {
+		const selectedDesign = useSelect(
+			( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getSelectedDesign(),
+			[]
+		);
+		const { setSelectedDesign, setIntent } = useDispatch( ONBOARD_STORE );
+		const selectedTheme = DEFAULT_ASSEMBLER_DESIGN.slug;
+		const theme = useSelector( ( state ) => getTheme( state, 'wpcom', selectedTheme ) );
+
+		// We have to query theme for the Jetpack site.
+		useQueryTheme( 'wpcom', selectedTheme );
+
+		useEffect( () => {
+			if ( ! theme ) {
+				// eslint-disable-next-line no-console
+				console.log( `The ${ selectedTheme } theme is loading...` );
+				return;
+			}
+
+			setSelectedDesign( {
+				...selectedDesign,
+				slug: theme.id,
+				title: theme.name,
+				recipe: {
+					...selectedDesign?.recipe,
+					stylesheet: theme.stylesheet,
+				},
+				design_type: 'assembler',
+			} );
+
+			setIntent( SiteIntent.WithThemeAssembler );
```

Comment:
@arthur791004 do you think we need to differentiate the intent `with-theme-assembler` and `assembler-first`?

Reply:
Oh yes, it should be `assembler-first` 🙈